### PR TITLE
datapath: Switch to upstream bpftool, remove additional arguments 

### PR DESCRIPTION
**Pull request by vadorovsky**
Upstream bpftool does not run probes which emit dmesg messages by
default anymore. Additional arguments for filtering out probes are not
needed anymore.

Link: https://lore.kernel.org/bpf/20200226165941.6379-1-mrostecki@opensuse.org/T/

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

```release-note
Switch to upstream bpftool
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10353)

### CODE REVIEW

**Comment by mrostecki** [comment]
@borkmann is going to submit a change in packer-ci-build repository which will switch bpftool to bpf-next tree this week.

But anyway, accepting this PR now should be fine. Even if Vagrant images are shipping the old bpftool now, this pull request should **still work**, although it will emit the dmesg warnings. But it's master, so it should be fine. And after the new VM image will ship the newest bpftool, dmesg warnings will be not emitted anymore.

 /cc @qmonnet @joestringer

**Comment by mrostecki** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]
LGTM. Any reservations on backporting into 1.7.1 so we have an official bpftool version in the main images for the latest release?

**Comment by mrostecki** [comment]
> LGTM. Any reservations on backporting into 1.7.1 so we have an official bpftool version in the main images for the latest release?

I think we need new Vagrant images with upstream bpftool before we backport it to any stable release. So we need to wait for @borkmann's PR in packer-ci-build.

File: pkg/datapath/linux/probes/probes.go
Comment (qmonnet) [inline pkg/datapath/linux/probes/probes.go:137]: Hmm maybe keep `probe`? It's more explicit if other subcommands are added to `feature` one day.

**Review by qmonnet — CHANGES_REQUESTED** [review]
One minor nit on bpftool invocation, looks good otherwise! Thanks.

File: contrib/packaging/docker/Dockerfile.runtime
Comment (aanm) [inline contrib/packaging/docker/Dockerfile.runtime]: It failed to build this image on quay:

```
2/27/2020, 10:06:34 AMsed "s|@SYSCONFDIR@|/etc/iproute2|g" ip-route.8.in > ip-route.8
2/27/2020, 10:06:34 AMfatal: Too many arguments. usage: git clone [<options>] [--] <repo> [<dir>]
2/27/2020, 10:06:34 AM
2/27/2020, 10:06:34 AM
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-v
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--verbose
2/27/2020, 10:06:35 AMbe more verbose
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-q
2/27/2020, 10:06:35 AM, --quiet
2/27/2020, 10:06:35 AMbe more quiet
2/27/2020, 10:06:35 AM--progress
2/27/2020, 10:06:35 AMforce progress reporting
2/27/2020, 10:06:35 AM-n
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--no-checkout
2/27/2020, 10:06:35 AMdon't create a checkout
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--bare
2/27/2020, 10:06:35 AMcreate a bare repository
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--mirror
2/27/2020, 10:06:35 AMcreate a mirror repository (implies bare)
2/27/2020, 10:06:35 AM-l
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--local
2/27/2020, 10:06:35 AMto clone from a local repository
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--no-hardlinks
2/27/2020, 10:06:35 AMdon't use local hardlinks, always copy
2/27/2020, 10:06:35 AM-s
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--shared
2/27/2020, 10:06:35 AMsetup as shared repository
2/27/2020, 10:06:35 AM--recurse-submodules
2/27/2020, 10:06:35 AM[=<pathspec>]
2/27/2020, 10:06:35 AMinitialize submodules in the clone
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-j
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--jobs
2/27/2020, 10:06:35 AM<n>
2/27/2020, 10:06:35 AMnumber of submodules cloned in parallel
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--template
2/27/2020, 10:06:35 AM<template-directory>
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AMdirectory from which templates will be used
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--reference
2/27/2020, 10:06:35 AM<repo>
2/27/2020, 10:06:35 AMreference repository
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--reference-if-able
2/27/2020, 10:06:35 AM<repo>
2/27/2020, 10:06:35 AMreference repository
2/27/2020, 10:06:35 AM--dissociate
2/27/2020, 10:06:35 AMuse --reference only while cloning
2/27/2020, 10:06:35 AM-o
2/27/2020, 10:06:35 AM, --origin
2/27/2020, 10:06:35 AM<name>
2/27/2020, 10:06:35 AMuse <name> instead of 'origin' to track upstream
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-b,
2/27/2020, 10:06:35 AM--branch
2/27/2020, 10:06:35 AM<branch>
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AMcheckout <branch> instead of the remote's HEAD
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-u
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--upload-pack
2/27/2020, 10:06:35 AM<path>
2/27/2020, 10:06:35 AMpath to git-upload-pack on the remote
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--depth
2/27/2020, 10:06:35 AM<depth>
2/27/2020, 10:06:35 AMcreate a shallow clone of that depth
2/27/2020, 10:06:35 AM--shallow-since
2/27/2020, 10:06:35 AM<time>
2/27/2020, 10:06:35 AMcreate a shallow clone since a specific time
2/27/2020, 10:06:35 AM--shallow-exclude
2/27/2020, 10:06:35 AM<revision>
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AMdeepen history of shallow clone, excluding rev
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--single-branch clone only one branch, HEAD or --branch
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--no-tags
2/27/2020, 10:06:35 AMdon't clone any tags, and make later fetches not to follow them
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM--shallow-submodules
2/27/2020, 10:06:35 AMany cloned submodules will be shallow
2/27/2020, 10:06:35 AM--separate-git-dir <gitdir>
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AMseparate git dir from working tree
2/27/2020, 10:06:35 AM-c
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--config
2/27/2020, 10:06:35 AM<key=value>
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AMset config inside the new repository
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-4
2/27/2020, 10:06:35 AM,
2/27/2020, 10:06:35 AM--ipv4
2/27/2020, 10:06:35 AMuse IPv4 addresses only
2/27/2020, 10:06:35 AM
2/27/2020, 10:06:35 AM-6
2/27/2020, 10:06:35 AM, --ipv6
2/27/2020, 10:06:35 AMuse IPv6 addresses only
2/27/2020, 10:06:35 AM--filter
2/27/2020, 10:06:35 AM<args>
2/27/2020, 10:06:35 AMobject filtering
2/27/2020, 10:06:35 AM
```

It's mostly likely related with the removal of ` && \`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by mrostecki** [comment]
@aanm Sorry for that! Also the dir name was wrong. Now it should work.

**Comment by mrostecki** [comment]
test-me-please

**Review by aanm — CHANGES_REQUESTED** [review]
@mrostecki please add the following changes to the `docker: Switch to upstream bpftool` commit

```diff
diff --git a/Dockerfile b/Dockerfile
index 7699103b7..1c724d074 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -37,7 +37,7 @@ RUN make LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 V=$V LIBNETWORK_PLUGIN=$LIBNETWORK_PLU
 # built while allowing the new versions to make changes that are not
 # backwards compatible.
 #
-FROM quay.io/cilium/cilium-runtime:2020-02-26
+FROM quay.io/cilium/cilium-runtime:2020-02-27
 LABEL maintainer="maintainer@cilium.io"
 COPY --from=builder /tmp/install /
 COPY --from=cilium-envoy / /
```

Once this PR is merged and PR #10354 is also merged, we need to open a PR against 1.7 for which the `cilium-runtime` image tag will be `2020-02-27-v1.7`

**Comment by mrostecki** [comment]
test-me-please

**Review by qmonnet — APPROVED** [review]

**Comment by mrostecki** [comment]
@aanm 
> Once this PR is merged and PR #10354 is also merged, we need to open a PR against 1.7 for which the `cilium-runtime` image tag will be `2020-02-27-v1.7`

We also need a change in packer-ci-build before doing that, but again, on sig-datapath meeting we decided that Daniel will do that alongside with the other changes he's preparing.

**Comment by aanm** [comment]
> @aanm
> 
> > Once this PR is merged and PR #10354 is also merged, we need to open a PR against 1.7 for which the `cilium-runtime` image tag will be `2020-02-27-v1.7`
> 
> We also need a change in packer-ci-build before doing that, but again, on sig-datapath meeting we decided that Daniel will do that alongside with the other changes he's preparing.

@mrostecki in that case, shouldn't we wait for the VM to be made available? Otherwise the cilium build will fail for old VMs, no?

**Comment by aanm** [comment]
@mrostecki 
```
17:54:01  Step 2/20 : FROM quay.io/cilium/cilium-builder:2020-02-27 as builder
17:54:02  manifest for quay.io/cilium/cilium-builder:2020-02-27 not found: manifest unknown: manifest unknown
```

the builder is not required, it should be the `runtime` image only.

**Review by jrajahalme — APPROVED** [review]
Looks good apart from the builder vs. runtime issue @aanm pointed out.

**Comment by mrostecki** [comment]
@aanm @jrajahalme Sorry for that, now it should work.

@aanm 
> @mrostecki in that case, shouldn't we wait for the VM to be made available? Otherwise the cilium build will fail for old VMs, no?

Well, the thing is that this pull request will work on old VMs (in terms of not being completely broken or making the CI red :wink: ), but will generate dmesg messages, which I guess is not a big deal on master branch (although it would be a big deal in stable branches).

That's because this PR goes back to the default set of arguments for `bpftool` and the upstream change is about not running `bpf_write_user` and `bpf_trace_printk` helper probes by default (if someone doesn't mind dmesg warnings and wants to run them, the additional argument `full` needs to be provided).

But on the other hand, I don't mind waiting for the change in packer-ci-build. I'm fine doing it either way. Maybe it's better to wait for the new VM to not confuse people and start backporting right after this PR is merged.

**Comment by mrostecki** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**
